pkg/subctl/cmd/join.go

```diff
@@ -268,16 +270,28 @@ func joinSubmarinerCluster(config *rest.Config, subctlData *datafile.SubctlData)
 		err = checkOverlappingClusterCidr(globalNetworks)
 		status.End(err == nil)
 		exitOnError("Error validating overlapping ClusterCIDRs", err)
-	} else if globalNetworks[clusterID] == nil || globalNetworks[clusterID].GlobalCIDRs == nil || len(globalNetworks[clusterID].GlobalCIDRs) <= 0 {
-		// Globalnet enabled, no globalCidr configured on this cluster
-		globalCIDR, err = globalnet.AllocateGlobalCIDR(globalNetworks, subctlData)
-		status.End(err == nil)
-		status.QueueSuccessMessage(fmt.Sprintf("Allocated GlobalCIDR: %s", globalCIDR))
-		exitOnError("Globalnet failed", err)
+	} else if globalCIDR == "" {
+		// Globalnet enabled, GlobalCIDR not specified by the user
+		if cidrispreconfigured(clusterID, globalNetworks) {
+			// globalCidr already configured on this cluster
+			globalCIDR = globalNetworks[clusterID].GlobalCIDRs[0]
+			status.QueueSuccessMessage(fmt.Sprintf("Cluster already has GlobalCIDR allocated: %s", globalNetworks[clusterID].GlobalCIDRs[0]))
+		} else {
+			// no globalCidr configured on this cluster
+			globalCIDR, err = globalnet.AllocateGlobalCIDR(globalNetworks, subctlData)
+			status.End(err == nil)
+			status.QueueSuccessMessage(fmt.Sprintf("Allocated GlobalCIDR: %s", globalCIDR))
+			exitOnError("Globalnet failed", err)
+		}
 	} else {
-		// Globalnet enabled, globalCidr already configured on this cluster
-		globalCIDR = globalNetworks[clusterID].GlobalCIDRs[0]
-		status.QueueSuccessMessage(fmt.Sprintf("Cluster already has GlobalCIDR allocated: %s", globalNetworks[clusterID].GlobalCIDRs[0]))
+		// Globalnet enabled, globalCIDR specified by user
+		if cidrispreconfigured(clusterID, globalNetworks) {
+			// globalCidr pre-configured on this cluster
+			globalCIDR = globalNetworks[clusterID].GlobalCIDRs[0]
+			status.QueueFailureMessage(fmt.Sprintf("Pre-configured GlobalCIDR %s detected. Not changing it.", globalCIDR))
```

Comment:
Please change the names in the debug/log messages to match the new flag (i.e., instead of GlobalCIDR, use globalnet-cidr to avoid confusion to the user)

Reply:
GlobalCIDR has been used since before so I didnt change it.

Reply:
Agree with Sridhar. If there is error, user needs to know which input parameter caused error. It was fine earlier because there was no user parameter for globalcidr, it was an internal variable so it was more for debugging than informing user. But now, we should tell the exact parameter that was passed wrong.

Reply:
Done